job-dsl-core/src/main/docs/examples/javaposse/jobdsl/dsl/Folder/icon.groovy

```diff
@@ -0,0 +1,28 @@
+// use the stock icon
+folder('stock')
+
+// use https://github.com/jenkinsci/custom-folder-icon-plugin for custom icons
+userContent('customFolderIcons/custom.png', streamFileFromWorkspace('custom.png'))
+folder('custom') {
+    icon {
+        customFolderIcon {
+            foldericon('custom.png')
+        }
+    }
+}
+
+// use https://github.com/jenkinsci/custom-folder-icon-plugin for ionicons
+folder('ionicon') {
```

Comment:
@strangelookingnerd Should we also have a test for new emoji type icons ?

Reply:
`job-dsl-plugin` requires Jenkins `2.361.4` whereas `custom-folder-icon-plugin` `2.5` (which includes the emoji icons) requires `2.387.1` - so I guess this won't be possible right now.

Reply:
I can probably get a release out next week. Once that's done I would have no problem with revving the minimum Jenkins core to 2.387.1.

Reply:
@basil That would be great! On a side-note I found that the tests in `javaposse.jobdsl.dsl.doc.ExampleValidationSpec` do not seem to be run during the build anymore or are at least not reporting any failing tests. Running them manually however shows that the test for my `icon.groovy` example still fails. Before the switch to Maven the behavior was different. Any idea why that is?

Reply:
I think it has something to do with the use of `@Unroll` with `file << findAllExamples()`, where the right-hand side is a function call rather than a static set of values. Seems that JUnit isn't able to invoke the function call correctly. Are you interested in opening a PR that removes the use of `@Unroll` in favor of a simple `for` loop inside of the test?

Reply:
#1279